com.unity.netcode.gameobjects/Runtime/Core/NetworkManager.cs

```diff
@@ -485,6 +586,399 @@ private void OnValidate()
+                        InvokeOnStartupFailedCallback(StartupFailureReason.BindFailed);
+                        ShutdownInternal();
+                        return false;
+                    }
+                }
+
+                if (IsHost)
+                {
+                    if (NetworkConfig.ConnectionApproval)
+                    {
+                        InvokeConnectionApproval(NetworkConfig.ConnectionData, ServerClientId,
+                            (createPlayerObject, playerPrefabHash, approved, position, rotation) =>
+                            {
+                                // You cannot decline the local server. Force approved to true
+                                if (!approved)
+                                {
+                                    if (NetworkLog.CurrentLogLevel <= LogLevel.Normal)
+                                    {
+                                        NetworkLog.LogWarning(
+                                            "You cannot decline the host connection. The connection was automatically approved.");
+                                    }
+                                }
+
+                                HandleApproval(ServerClientId, createPlayerObject, playerPrefabHash, true, position, rotation);
+                            });
+                    }
+                    else
+                    {
+                        HandleApproval(ServerClientId, NetworkConfig.PlayerPrefab != null, null, true, null, null);
+                    }
+                }
+
+                if (IsServer)
+                {
+                    SpawnManager.ServerSpawnSceneObjectsOnStartSweep();
+                    OnServerStarted?.Invoke();
+                }
+            }
+
+            return true;
+        }
+
+        public void AddNetworkPrefab(GameObject prefab)
+        {
+            if (State != NetworkManagerState.Inactive)
+            {
+                throw new Exception($"All prefabs must be registered before starting {nameof(NetworkManager)}");
+            }
+
+            NetworkPrefab.VerifyValidPrefab(prefab);
+            NetworkConfig.NetworkPrefabs.Add(new NetworkPrefab { Prefab = prefab });
+        }
+
+#if NETCODE_USE_ADDRESSABLES
+        public void AddNetworkPrefab(AssetReferenceGameObject addressableAsset)
+        {
+            if (State != NetworkManagerState.Inactive)
+            {
+                throw new Exception($"All prefabs must be registered before starting {nameof(NetworkManager)}");
+            }
+
+            NetworkConfig.NetworkAddressables.Add(new NetworkAddressable { Addressable = addressableAsset });
+        }
+        public void AddNetworkPrefab(string address)
+        {
+            AddNetworkPrefab(new AssetReferenceGameObject(address));
+        }
+
+        public GameObject GetGameObjectForAddressable(AssetReferenceGameObject addressableAsset)
```

Comment:
What would the use case for this public API be?  Do we need to expose this?

Reply:
Just trying to make it a little easier for users to use the API. I find the addressables API a little clunky and difficult to understand so I thought having something like this might make it easier - since we're doing all the rest of the addressables API under the hood, just felt to me like it was worthwhile to let the user just not think about the addressables API at all if they don't want to. Not opposed to removing it if you think that's superfluous.